daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c

```diff
@@ -1520,8 +1545,48 @@ static int ipapwd_pre_bind(Slapi_PBlock *pb)
     /* Attempt to write out kerberos keys for the user. */
     ipapwd_write_krb_keys(pb, discard_const(dn), entry, credentials);
 
+    /* Get the (user's) uid and $SUFFIX parts of the DN. */
+    uid_len = (size_t)(strchr(dn, ',') - dn);
+    strncpy(uid, dn, uid_len);
+    uid[uid_len] = '\0';
+    suffix_start = strstr(dn, ",dc=");
+    suffix_len = strlen(dn) - (size_t)(suffix_start - dn);
+    suffix = malloc(suffix_len + 1);
+    strncpy(suffix, suffix_start, suffix_len);
```

Comment:
A more robust way to do this would be slapi_ldap_explode_dn() to break the DN into RDNs and operate on those.

Reply:
I heeded this advice here as well, the same holds as mentioned [above](https://github.com/freeipa/freeipa/pull/4923#discussion_r460436550).